src/ray/object_manager/object_manager.cc

```diff
@@ -181,8 +184,10 @@ void ObjectManager::NotifyDirectoryObjectDeleted(const ObjectID &object_id) {
   local_objects_.erase(it);
   used_memory_ -= object_info.data_size + object_info.metadata_size;
   RAY_CHECK(!local_objects_.empty() || used_memory_ == 0);
-  ray::Status status =
-      object_directory_->ReportObjectRemoved(object_id, self_node_id_, object_info);
+  if (!is_object_spilled_locally_(object_id)) {
```

Comment:
Without this, remote nodes will never be able to find that this node has an object because the location is removed when the object is evicted.